pkg/build/api/v1/types.go

```diff
@@ -371,6 +375,13 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// RunBash holds a bash script to be executed in a container running the
+	// build output image.
+	RunBash string `json:"runBash,omitempty" description:"bash script to be executed in a container running the build output image"`
```

Comment:
@bparees +1 `--post-commit-cmd="rake test"` and `--post-commit-entrypoint="/bin/bash"` (if you need to). 

the UI can alternatively show two field, but the entrypoint field hidden by default as "advanced something". Also I don't see problems constructing slice from string unless you have to deal with spaces in quotes, etc...